legend-engine-application-query/src/main/java/org/finos/legend/engine/application/query/api/QueryStoreManager.java

```diff
@@ -239,11 +241,20 @@ public List<Query> getQueries(QuerySearchSpecification searchSpecification, Stri
                     ListIterate.collect(searchSpecification.stereotypes, stereotype ->
                             Filters.and(Filters.eq("stereotypes.profile", stereotype.profile), Filters.eq("stereotypes.value", stereotype.value)))));
         }
-        return LazyIterate.collect(this.getQueryCollection()
-                .find(filters.isEmpty() ? EMPTY_FILTER : Filters.and(filters)).sort(searchSpecification.showLatestQueriesFirst != null && searchSpecification.showLatestQueriesFirst ? descending("lastUpdatedAt") : EMPTY_FILTER)
-                // NOTE: return a light version of the query to save bandwidth
-                .projection(Projections.include("id", "name", "versionId", "groupId", "artifactId", "owner", "createdAt", "lastUpdatedAt"))
-                .limit(Math.min(MAX_NUMBER_OF_QUERIES, searchSpecification.limit == null ? Integer.MAX_VALUE : searchSpecification.limit)), QueryStoreManager::documentToQuery).toList();
+        filters.add(Filters.in("owner", currentUser, null));
+        List<Query> currentUserQueryResults = this.getQuerySearchResults(searchSpecification, filters);
+        if (searchSpecification.showCurrentUserQueriesOnly != null && searchSpecification.showCurrentUserQueriesOnly)
+        {
+            return currentUserQueryResults;
+        }
+        filters.remove(Filters.in("owner", currentUser, null));
+        filters.add(Filters.nin("owner", currentUser, null));
+        searchSpecification.limit = searchSpecification.limit != null ? searchSpecification.limit - currentUserQueryResults.size() : Integer.MAX_VALUE - currentUserQueryResults.size();
```

Comment:
this looks hacky, say if we remove `showLatestQueriesFirst`, the only thing left from search specification that `getQuerySearchResults()` needs is the limit, we can just send in the limit and the filters